yo1k/tic_tac_toe/core/util.py

```diff
@@ -1,6 +1,12 @@
-def eq(cls):
+from __future__ import annotations
+from typing import TypeVar, Type
+
+T = TypeVar("T")  # pylint: disable=C0103
+
+
+def eq(cls: Type[T]) -> Type[T]:
     """Class decorator providing generic comparison functionality."""
-    def __eq__(self, other) -> bool:
+    def __eq__(self: T, other: object) -> bool:
         return isinstance(other, self.__class__) and self.__dict__ == other.__dict__
-    cls.__eq__ = __eq__
+    cls.__eq__ = __eq__  # type: ignore
```

Comment:
`Mypy` do not allow assigning a function to a method (see [here](https://github.com/python/mypy/issues/2427)).